src/sentry/api/endpoints/project_searches.py

```diff
@@ -17,6 +17,8 @@ class SavedSearchSerializer(serializers.Serializer):
 
 
 class ProjectSearchesEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectSearchPermission,)
```

Comment:
@dcramer as suggested by you I have tried to implement allowing member level saved search. I have tried to follow above points. 

Wasn't sure about following 
=> private searches show up differently in the UI to suggest they're private
Should I add P label aligned with private searches to show that they are private?